src/core.ts

```diff
@@ -0,0 +1,327 @@
+import { ApolloClient, InMemoryCache, NormalizedCacheObject } from "@apollo/client/core";
+import { createClient } from "graphql-ws";
+import { GraphQLWsLink } from "@apollo/client/link/subscriptions";
+import { Ad4mClient } from "@perspect3vism/ad4m";
+import { checkPort } from "./utils";
+
+if (window === undefined) {
+  console.log('Running ad4m-connect in node.js');
+} else {
+  console.log('Running ad4m-connect in browser');
+}
+
+
+function runNode() {
+  
+}
+
+function runBrowser(args: Ad4mConnectOptions): Client {
+  const port = args.port || parseInt(localStorage.getItem("ad4minPort") || "12000");
+  const token = args.token || localStorage.getItem("ad4minToken") || "";
+  const url = args.url || localStorage.getItem("ad4minURL") || "";
+
+  const setExecutorUrl = (url: string) => {
+    localStorage.setItem("ad4minURL", url);
+  }
+
+  const setExecutorPort = (port: string) => {
+    localStorage.setItem("ad4minPort", port);
+  }
+
+  const setExecutorToken = (token: string) => {
+    localStorage.setItem("ad4minToken", token);
+  }
+
+  const client = new Client({
+    ...args,
+    port,
+    token,
+    url,
+    setExecutorUrl: args.setExecutorUrl || setExecutorUrl,
+    setExecutorPort: args.setExecutorPort || setExecutorPort,
+    setExecutorToken: args.setExecutorToken || setExecutorToken
+  });
+
+  return client;
+}
+
+type Ad4mConnectOptions = {
```

Comment:
Some of the options only depends when use with electron.
I'm thinking unify the electron and web will simplify a lot of the codes.
On way to do the unify is let electron search port and store cap token in localStorage.